core/src/main/resources/org/apache/spark/ui/static/utils.js

```diff
@@ -205,6 +205,29 @@ function createRESTEndPointForExecutorsPage(appId) {
   return uiRoot + "/api/v1/applications/" + appId + "/allexecutors";
 }
 
+function createRESTEndPointForExecutorsPeakMetricsSummariesPage(appId) {
+  var words = getBaseURI().split('/');
```

Comment:
It seems like most of the logic here is duplicated with `createRESTEndPointForExecutorsPage` except for the REST call. Can we refactor it a bit and avoid/reduce the redundant code?

Reply:
> It seems like most of the logic here is duplicated with `createRESTEndPointForExecutorsPage` except for the REST call. Can we refactor it a bit and avoid/reduce the redundant code?

It belongs to different api,  `allexecutors` return the list of all executor's summaries. It's hard to combine `/allexecutors` and `/executorPeakMemoryMetricsDistribution `

Reply:
How about current.